### DISCUSSION TITLE
psycopg2 vs psycopg2-binary

### DISCUSSION THREAD

**Discussion by brki** [apache/airflow#25677] [General]
Airflow's [Postgres provider](https://github.com/apache/airflow/blob/main/airflow/providers/postgres/provider.yaml) specifies _psycopg2-binary_.

According to the [psycopg2 docs](https://www.psycopg.org/docs/install.html#psycopg-vs-psycopg-binary), this is not recommended for production use : 

> The psycopg2-binary package is meant for beginners to start playing with Python and PostgreSQL without the need to meet the build requirements.
> 
> If you are the maintainer of a published package depending on psycopg2 you shouldn’t use psycopg2-binary as a module dependency. For production use you are advised to use the source distribution.

Shouldn't Airflow use the non-binary package _psycopg2_?

I guess some people will point out that it requires additional effort to build, because the OS-level PostgreSQL dev libraries, and openssl libraries, need to be available.

If the goal is to have Airflow easily installable, perhaps there should be an alternative postgres provider, that has a dependency on the non-binary _psycopg2_ package?

**Comment by potiuk** [comment]
I think you are right - can you please make a PR and see if you can make the tests and images work? Shoudl be as easy as updating `provider.yaml` dependency. Creating a PR should run all tests and we should see if we can swap it out - it's really the matter if our image will nicely build with it,

**Reply by brki** [reply]
PR: #25710
